modules/dashboard/php/endpoints/dashboard.class.inc

```diff
@@ -0,0 +1,570 @@
+ * The NDB_Page for the POST requests.
+ *
+ * @category Loris
+ * @package  Dashboard
+ * @author   Tara Campbell <tara.campbell@mail.mcgill.ca>
+ *           Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class Dashboard extends Endpoint implements ETagCalculator
+{
+    public $skipTemplate = true;
+    /**
+     * This function will return a json response.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Ensure GET or POST request.
+        switch ($request->getMethod()) {
+        case 'GET':
+            return $this->_handleGET($request);
+        default:
+            return new \LORIS\Http\Response\MethodNotAllowed(
+                $this->allowedMethods()
+            );
+        }
+    }
+
+    /**
+     * Initialize setup, the extra values for the
+     * create timepoint form. (psc & errors)
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleGET(ServerRequestInterface $request) : ResponseInterface
+    {
+        $response = $this->initializeSetup();
+
+        return new \LORIS\Http\Response\JsonResponse(
+            $response
+        );
+    }
+
+    /**
+     * Initialize setup, building the Dashboard.
+     *
+     * @return array
+     * @throws \LorisException
+     */
+    function initializeSetup() : array
+    {
+        // data for panels.
+        $welcomeData            = array();
+        $recruitmentData        = array();
+        $studyProgressionData   = array();
+        $myTasksData            = array();
+        $documentRepositoryData = array();
+        $behaviouralFeedbackData = array();
+
+        $factory = \NDB_Factory::singleton();
+        $DB      = $factory->database();
+        $user    = $factory->user();
+        $config  = $factory->config();
+        $site    = $user->getSiteNames();
+
+        $userID    = $user->getUsername();
+        $lastLogin = $DB->pselectOne(
+            "SELECT MAX(Login_timestamp)
+             FROM user_login_history
+             WHERE Login_timestamp < (SELECT MAX(Login_timestamp)
+                FROM user_login_history
+                WHERE userID=:UserID AND Success='Y')
+             AND userID=:UserID AND Success='Y'",
+            array('UserID' => $userID)
+        );
+
+        $siteID = $user->getCenterIDs();
+        // todo user_site used with myTasks
+        $myTasksData['user_site'] = $siteID;
+
+        // Welcome panel
+        // todo username used with welcome
+        $welcomeData['username'] = $user->getFullname();
+        if ($lastLogin) {
+            // todo lastLogin used with welcome
```

Comment:
I'm not sure I understand this. If you didn't write this and this is copied from another author, please delete it.

Reply:
@johnsaigle I didn't write this but I know that this code is for altering the welcome message. Such as when it's the very first time the user has logged in and the times afterwards.. So I don't think it should be deleted. I'll remove the todo comments.